glean-core/ios/Glean/Dispatchers.swift

```diff
@@ -176,6 +185,13 @@ class Dispatchers {
         // Turn off queuing to allow for normal background execution mode
         queueInitialTasks.value = false
 
+        // This must happen after `queueInitialTasks.set(false)` is run, or it
+        // would be added to a full task queue and be silently dropped.
+        if preInitTaskCount > Constants.maxQueueSize {
+            GleanMetrics.GleanError.preinitTasksOverflow.add(preInitTaskCount)
+        }
+        preInitTaskCount = 0
```

Comment:
Just out of curiosity, why do you need zero it out here again?

Reply:
Mostly for tests which don't necessarily destroy the copy of the `MetricsPingScheduler` held by Glean.  I'll add a comment to help clarify why it was done.